conda/core/repodata.py

```diff
@@ -413,7 +413,7 @@ def fetch_repodata(url, schannel, priority,
         mtime = getmtime(cache_path)
     except (IOError, OSError):
         log.debug("No local cache found for %s at %s", url, cache_path)
-        if use_cache:
+        if context.offline or use_cache:
```

Comment:
As opposed to https://github.com/conda/conda/pull/5134#discussion_r113264684, this is definitely going to be a problem as it is right now.  Line 431 is

    if (timeout > 0 or context.offline) and not url.startswith('file://'):

For the `--offline` flag, we need to still let urls that start with `file://` pass through to requests, which will use `conda/gateways/adapters/localfs.py` to "fetch" the repodata for the channel while still using all the same basic logic as remote requests.